content/advanced/340_appmesh_flagger/flagger_setup.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Flagger Set Up"
+date: 2021-03-18T00:00:00-05:00
+weight: 10
+draft: false
+---
+
+#### PreRequisite
+
+We assume that we have already have the following setup before we start this chapter.
+  * [**Required**] An existing EKS Cluster `eksworkshop-eksctl` created from [EKS Workshop](/030_eksctl/launcheks/)
+  * [**Required**] [Nodegroup setup](/advanced/330_servicemesh_using_appmesh/add_nodegroup_fargate/create_nodegroup/) completed from AppMesh Workshop.
+  * [**Required**] [AppMesh setup](/advanced/330_servicemesh_using_appmesh/appmesh_installation/install_appmesh/) completed from AppMesh Workshop.
+  * [**Required**] [increased the disk size on your Cloud9 instnace](020_prerequisites/workspace/#increase-the-disk-size-on-the-cloud9-instance) as we need to build container images for our application.
+
+
+ #### Install Flagger
+
+Add Flagger Helm repository:
+
+```bash
+helm repo add flagger https://flagger.app
+```
+
+{{< output >}}
+"flagger" has been added to your repositories
+{{< /output >}}
+
+Install Flagger's Canary CRD:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/weaveworks/flagger/master/artifacts/flagger/crd.yaml
+```
+{{< output >}}
+customresourcedefinition.apiextensions.k8s.io/canaries.flagger.app created
+customresourcedefinition.apiextensions.k8s.io/metrictemplates.flagger.app created
+customresourcedefinition.apiextensions.k8s.io/alertproviders.flagger.app created
+{{< /output >}}
+
+Deploy Flagger in the appmesh-system namespace:
+
+```bash
+helm upgrade -i flagger flagger/flagger \
+--namespace=appmesh-system \
+--set crd.create=false \
+--set meshProvider=appmesh:v1beta2 \
+--set metricsServer=http://appmesh-prometheus:9090
+```
+{{< output >}}    
+Release "flagger" does not exist. Installing it now.
+NAME: flagger
+LAST DEPLOYED: Mon Dec 14 15:23:32 2020
+NAMESPACE: appmesh-system
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+Flagger installed
+{{< /output >}}
+
+
+#### Install the App Mesh Prometheus
+
+```bash
+helm upgrade -i appmesh-prometheus eks/appmesh-prometheus \
+--wait --namespace appmesh-system
+```
+
+{{< output >}}
+Release "appmesh-prometheus" does not exist. Installing it now.
+NAME: appmesh-prometheus
+LAST DEPLOYED: Sat Mar 13 20:59:29 2021
+NAMESPACE: appmesh-system
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+AWS App Mesh Prometheus installed!
+{{< /output >}}
+
+#### Enable horizontal pod auto-scaling
```

Comment:
May be we can change the heading to "Set up the horizontol pod autoscaler"

Reply:
Done